From 71e3377a6e56747855a7d6f3c8b04cf590bd6cd4 Mon Sep 17 00:00:00 2001
From: Reilly Brogan <reilly@reillybrogan.com>
Date: Tue, 12 Dec 2023 18:42:47 -0600
Subject: [PATCH] Load default openssl provider

Qt 6.6.1 changed it so that the default openssl provider is no longer loaded by default. We need that for the grpc connection to the bridge CLI, so load it manually during init.

This is probably not idiomatic in the context of Qt but it does appear to work and it matches the OpenSSL examples.

Ref: https://github.com/getsolus/packages/issues/996

Signed-off-by: Reilly Brogan <reilly@reillybrogan.com>
---
 internal/frontend/bridge-gui/bridge-gui/main.cpp | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/internal/frontend/bridge-gui/bridge-gui/main.cpp b/internal/frontend/bridge-gui/bridge-gui/main.cpp
index dc8eca14..38db9f6b 100644
--- a/internal/frontend/bridge-gui/bridge-gui/main.cpp
+++ b/internal/frontend/bridge-gui/bridge-gui/main.cpp
@@ -29,6 +29,7 @@
 #include <bridgepp/Log/Log.h>
 #include <bridgepp/Log/LogUtils.h>
 #include <bridgepp/ProcessMonitor.h>
+#include <openssl/provider.h>
 
 
 #ifdef Q_OS_MACOS
@@ -284,6 +285,13 @@ int main(int argc, char *argv[]) {
         QCoreApplication::setAttribute(Qt::AA_UseSoftwareOpenGL); // must be called before instantiating the BridgeApp
     }
 
+    OSSL_PROVIDER *deflt;
+    deflt = OSSL_PROVIDER_load(NULL, "default");
+    if (deflt == NULL) {
+        printf("Failed to load Default provider\n");
+        return EXIT_FAILURE;
+    }
+
     BridgeApp guiApp(argc, argv);
     initSentry();
     auto sentryCloser = qScopeGuard([] { sentry_close(); });
@@ -411,6 +419,7 @@ int main(int argc, char *argv[]) {
         }
         // release the lock file
         lock.unlock();
+        OSSL_PROVIDER_unload(deflt);
         return result;
     }
     catch (Exception const &e) {
-- 
2.43.0

